nilearn/input_data/masker_validation.py

```diff
@@ -0,0 +1,75 @@
+import warnings
+
+from .._utils.class_inspect import get_params
+from . import MultiNiftiMasker, NiftiMasker
+
+
+def check_embedded_nifti_masker(estimator, multi_subject=True):
+    """Base function for using a masker within a BaseEstimator class :
+    This creates a masker from instance parameters :
+    - If instance contains a mask image in mask parameter,
+    we use this image as new masker mask_img, forwarding instance parameters to
+    new masker : smoothing_fwhm, standardize, detrend, low_pass= high_pass,
+    t_r, target_affine, target_shape, mask_strategy, mask_args,
+    - If instance contains a masker in mask parameter, we use a copy of
+    this masker, overidding all instance masker related parameters.
+    In all case, we forward system parameters of instance to new masker :
+    memory, memory_level, vebose, n_jobs
+
+    Parameters
+    ==========
+        instance: object, instance of BaseEstimator
+            The object that gives us the values of the parameters
+
+        multi: boolean, default=True,
```

Comment:
I don't think you are supposed to end the line with a comma but I may be wrong about that.

Interestingly, looks like the numpy doc is encouraging to put optional in the type but the default value in the parameter description rather than the type, see https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections. I don't think that's something which we applied very broadly in nilearn though.